SpatialGDK/Source/SpatialGDK/Private/Interop/GlobalStateManager.cpp

```diff
@@ -595,10 +599,27 @@ void UGlobalStateManager::BecomeAuthoritativeOverAllActors()
 	}
 }
 
+void UGlobalStateManager::BecomeAuthoritativeOverActorsBasedOnLBStrategy()
+{
+	for (TActorIterator<AActor> It(NetDriver->World); It; ++It)
+	{
+		AActor* Actor = *It;
+		if (Actor != nullptr && !Actor->IsPendingKill())
+		{
+			if (Actor->GetIsReplicated() && NetDriver->LoadBalanceStrategy->ShouldRunBeginPlayWithAuthority(*Actor))
+			{
+				Actor->Role = ROLE_Authority;
+				Actor->RemoteRole = ROLE_SimulatedProxy;
+			}
+		}
+	}
+}
+
 void UGlobalStateManager::TriggerBeginPlay()
 {
 	check(GetCanBeginPlay());
 
+	BecomeAuthoritativeOverActorsBasedOnLBStrategy();
```

Comment:
Only do this if load balancer is enabled.

Reply:
Could you also add a comment that this function gets called by all servers - not just the one authoritative over the gsm?

Reply:
Refactored so all logic is in this function instead